internal/ent/hooks/assessment_response.go

```diff
@@ -0,0 +1,215 @@
+package hooks
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"entgo.io/ent"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/google/uuid"
+	"github.com/theopenlane/core/internal/ent/generated"
+	"github.com/theopenlane/core/internal/ent/generated/assessment"
+	"github.com/theopenlane/core/internal/ent/generated/assessmentresponse"
+	"github.com/theopenlane/core/internal/ent/generated/hook"
+	"github.com/theopenlane/core/internal/ent/generated/organization"
+	"github.com/theopenlane/core/internal/graphapi/gqlerrors"
+	"github.com/theopenlane/core/internal/httpserve/authmanager"
+	"github.com/theopenlane/core/pkg/enums"
+	"github.com/theopenlane/core/pkg/logx"
+	"github.com/theopenlane/emailtemplates"
+	"github.com/theopenlane/iam/tokens"
+	"github.com/theopenlane/riverboat/pkg/jobs"
+)
+
+// HookCreateAssessmentResponse sends the email to the user to fill in and input their data.
+// It also makes sure to bump up the send attempts if needed.
+func HookCreateAssessmentResponse() ent.Hook {
+	return hook.On(func(next ent.Mutator) ent.Mutator {
+		return hook.AssessmentResponseFunc(func(ctx context.Context, m *generated.AssessmentResponseMutation) (generated.Value, error) {
+			id, ok := m.AssessmentID()
+			email, emailExists := m.Email()
+
+			if !ok || !emailExists {
+				m.ClearDocumentDataID()
+				return next.Mutate(ctx, m)
+			}
+
+			if err := validateAndSetDueDate(ctx, m); err != nil {
+				return nil, err
+			}
+
+			existingResponse, err := m.Client().AssessmentResponse.Query().
+				Where(
+					assessmentresponse.AssessmentID(id),
+					assessmentresponse.Email(email),
+				).
+				Only(ctx)
+
+			if err != nil {
+				if !generated.IsNotFound(err) {
+					return nil, err
+				}
+
+				// not found so this is a new user
+				m.ClearDocumentDataID()
+
+				value, err := next.Mutate(ctx, m)
+				if err != nil {
+					return nil, err
+				}
+
+				if err := createResponseEmail(ctx, m); err != nil {
+					logx.FromContext(ctx).Error().Err(err).Msg("failed to send assessment response email")
+				}
+
+				return value, nil
+			}
+
+			// if started, we cannot send an invitation again
+			if existingResponse.Status != enums.AssessmentResponseStatusSent {
+				return nil, ErrAssessmentInProgress
+			}
+
+			newAttempts := existingResponse.SendAttempts + 1
+			if newAttempts > maxAttempts {
+				return nil, gqlerrors.NewCustomError(
+					gqlerrors.MaxAttemptsErrorCode,
+					"max attempts reached for this email, please delete the invite and try again",
```

Comment:
Not an invite - believe this would be a request to fill out a questionnaire?
```suggestion
					"max attempts reached for this email, please delete the questionnaire request and try again",
```